File: python/functions/file_operations.py
#!/usr/bin/env python3

import os

test_file = '../tmp/test_output.txt'

data = "Hello from Python!\nThis is a test file.\n"
with open(test_file, 'w') as file_handle:
    file_handle.write(data)
print(f"Written to {test_file}")

with open(test_file, 'r') as file_handle:
    content = file_handle.read()
print(f"Read content:\n{content}")

with open(test_file, 'r') as file_handle:
    lines = file_handle.readlines()
print(f"Lines: {lines}")

if os.path.exists(test_file):
    print(f"File exists: {test_file}")
    print(f"File size: {os.path.getsize(test_file)} bytes")

File: ruby/functions/file_operations.rb
#!/usr/bin/env ruby

test_file = '../tmp/test_output.txt'

data = "Hello from Ruby!\nThis is a test file.\n"
File.write(test_file, data)
puts "Written to #{test_file}"

content = File.read(test_file)
puts "Read content:\n#{content}"

lines = File.readlines(test_file)
puts "Lines: #{lines}"

if File.exist?(test_file)
  puts "File exists: #{test_file}"
  puts "File size: #{File.size(test_file)} bytes"
end

File.open(test_file, 'a') do |file|
  file.puts 'Appended line'
end
puts "Appended to file"

Dir.foreach('../tmp') do |filename|
  next if filename == '.' || filename == '..'
  puts "Found file: #{filename}"
end
